src/context/LogContext/state.ts

```diff
@@ -0,0 +1,42 @@
+import { useReducer } from "react";
+
+interface LogState {
+  logs: string[];
+  lineCount: number;
+}
+
+type Action = { type: "INGEST_LOGS"; logs: string[] } | { type: "CLEAR_LOGS" };
+
+const initialState: LogState = {
+  logs: [],
+  lineCount: 0,
+};
+
+const reducer = (state: LogState, action: Action) => {
+  switch (action.type) {
+    case "INGEST_LOGS":
+      return {
+        ...state,
+        logs: action.logs,
+        lineCount: action.logs.length,
```

Comment:
Do you think `lineCount` is necessary to manage via state? I would argue that because it is directly derived from `logs`, it's more of an intermediate value and would be more straightforward to export a function from the hook that is a log length getter than to manage separately.

Reply:
Good point done!